### ISSUE TITLE
setup.py install doesn't work on Mac OS X

### ISSUE THREAD

**Issue by jonnybaik** [ipython-contrib/IPython-notebook-extensions#431]
I was following the instructions to install the notebook extensions by following the instructions in the [README](https://github.com/ipython-contrib/IPython-notebook-extensions/blob/master/README.md). I checked out the master branch, then ran `python setup.py install`, but received the following error:

```
$ python setup.py install
Traceback (most recent call last):
  File "/usr/local/var/pyenv/versions/3.4.3/lib/python3.4/site-packages/psutil/_psosx.py", line 199, in wrapper
    return fun(self, *args, **kwargs)
  File "/usr/local/var/pyenv/versions/3.4.3/lib/python3.4/site-packages/psutil/_psosx.py", line 238, in cmdline
    return cext.proc_cmdline(self.pid)
ProcessLookupError: [Errno 3] No such process

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 24, in <module>
    if "jupyter-notebook" in p.name():
  File "/usr/local/var/pyenv/versions/3.4.3/lib/python3.4/site-packages/psutil/__init__.py", line 539, in name
    cmdline = self.cmdline()
  File "/usr/local/var/pyenv/versions/3.4.3/lib/python3.4/site-packages/psutil/__init__.py", line 592, in cmdline
    return self._proc.cmdline()
  File "/usr/local/var/pyenv/versions/3.4.3/lib/python3.4/site-packages/psutil/_psosx.py", line 209, in wrapper
    raise ZombieProcess(self.pid, self._name, self._ppid)
psutil.ZombieProcess: psutil.ZombieProcess process still exists but it's a zombie (pid=64)
```

It seems to crash when it checks to see if jupyter-notebook currently running.

My setup:

Mac OS X 10.11.1 (El Capitan)
Python 3.4.3, installed using pyenv

**Comment by jonnybaik** [comment]
The lines of code in `setup.py` causing the crash:

```
 for p in psutil.process_iter():
     if "jupyter-notebook" in p.name():
         print("Cannot install while the Jupyter notebook server is running")
         exit()
```

When I google the problem, I find this archived thread discussing how `psutil` not properly handling zombie processes in OS X, throwing an exception when it comes across a zombie process.

I solved the issue by handling the zombie process exception by doing nothing:

```
for p in psutil.process_iter():
    try:
        if "jupyter-notebook" in p.name():
            print("Cannot install while the Jupyter notebook server is running")
            exit()
    except psutil.ZombieProcess:
        pass
```

**Comment by jonnybaik** [comment]
Oh, another issue turned up. It seems that (on my setup, at least), `psutil` fails to find the jupyter-notebook process when checking just the process names.

I started the notebook server by running `jupyter notebook` in the terminal, and then tried running `python setup.py install`. To my surprise, it did not exit or complain that jupyter notebook was running. Looking at the code, it seems that my jupyter notebook is running with a process name of `python3.4`. 

To programmatically find which process was running the juptyer-notebook, I had to check the command that invoked the process, but this would throw a `AccessDenied` exception when trying to check processes that were not owned by the user. The code below gave me the desired behavior:

```{python}
for p in psutil.process_iter():
    try:
        if "jupyter-notebook" in p.name() or any("jupyter-notebook" in c for c in p.cmdline()):
            print("Cannot install while the Jupyter notebook server is running")
            exit()
    except (psutil.ZombieProcess, psutil.AccessDenied):
        pass
```
